File: 2-2/schedule.ml
let n = 5
let s = [1; 2; 4; 6; 8]
let t = [3; 5; 7; 9; 10]

let plst = List.sort compare (List.map2 (fun a b -> (b, a)) s t)

let time = ref 0

let solve = List.fold_left
    (fun n (p1, p2) -> if (!time < p2) then (time := p1; n + 1) else n) 0 plst

File: 2-2/schedule.cpp
#include <bits/stdc++.h>
using namespace std;

int n = 5;
vector<int> s = {1, 2, 4, 6, 8};
vector<int> t = {3, 5, 7, 9, 10};
vector<pair<int, int>> p(n);

int solve() {
  int ans = 0;
  int time = 0;
  for (int i=0; i<n; i++) {
    if (time < p.at(i).second) {
      time = p.at(i).first;
      ans ++;
    }
  }
  return ans;
}

int main() {

  for (int i=0; i<n; i++) {
    p.at(i).first = t.at(i);
    p.at(i).second = s.at(i);
  }
  sort(p.begin(), p.end());
  
  int ans = solve();
  cout << ans << "\n";
}
